engine/engine-paper/src/main/kotlin/com/typewritermc/engine/paper/interaction/ChatHistoryHandler.kt

```diff
@@ -56,12 +103,19 @@ class ChatHistoryHandler :
             val component = message.message
             val history = getHistory(event.user.uuid)
             if (component is TextComponent && component.content().startsWith("no-index")) {
-                if (component.content().endsWith("resend")) return
-
                 history.allowedMessageThrough()
                 return
             }
+
+            val player = server.getPlayer(event.user.uuid) ?: return
```

Comment:
I don't think it's useful to fetch the full player here to then only use the uuid of it again later. 

Just pass the uuid of the player in the token thing because then it also doesn't depend on the paper api